### ISSUE TITLE
Feature Request: slot inside the auth section

### ISSUE THREAD

**Issue by splitbrain** [Authress-Engineering/openapi-explorer#209]
Currently only the whole auth page can be replaced using the `authentication` slot. It would be great to have a slot to add content without losing the ability to add an API key.

**Comment by wparad** [comment]
Closing as this is a duplicate of #193. In that issue it explains how to do this, alternatives, and why this doesn't exist as a separate slot. If for some reason that doesn't work out, please do let us know.

**Comment by wparad** [comment]
Right now it is one line of javascript and the associated HTML that would need to be added. As the Auth page is vanilla, it hasn't made sense to attempt to add additional slots there. The example javascript snippet in #193, and should work out of the box. If you find a problem with that, it is something we would definitely want to improve.

The doc example for this is lean, but it is here: https://github.com/Authress-Engineering/openapi-explorer/blob/release/2.1/docs/documentation.md#library-api

I'll copy it here for brevity:
```js

**Comment by splitbrain** [comment]
The doc you link says:

> Set a token for methods that require security for a particular security scheme id.

But that's not what I want. I want to keep the ability for the user to provide their own credentials. I just want to add some explanations below:

![screenshot-localhost-2023 12 01-18_14_50](https://github.com/Authress-Engineering/openapi-explorer/assets/86426/a2f7b7be-3206-4905-81ad-32ed1b90279f)

From your answer I get that I could maybe use JavaScript to insert some HTML into the DOM generated by your component (though I would have expected that to be prevented by the component's shadow dom). However even if that is possible, it's not a route I want to go. It would be more code that needs to be maintained and I rather focus on maintaining my own project.

If adding a slot there is difficult or not in the scope of your project that's fine with me. I can also add my few words to the overview page. The authentication page would just have been a better fit.

Thanks for the super prompt responses btw!

**Comment by wparad** [comment]
Nah, I'm definitely not suggesting to dynamically insert html into the DOM.

What I am suggesting replace the Auth slot with your content. You can add exactly what you want. For the buttons/input text fields, you only need call the javascript line of code when your custom `SET` button is pressed.

for example, I just copied out the DOM for our public tester:

```html
<section id="auth" part="section-auth" class="observe-me section-gap--focused-mode">
   <slot name="authentication">
      <div class="section-padding">
         <div class="sub-title regular-font">AUTHENTICATION</div>
         <div class="small-font-size" style="display:flex;align-items:center;min-height:30px">
            <div class="red-text">No API key applied</div>
         </div>
         <table role="presentation" class="m-table" style="width:100%">
            <tr>
               <td colspan="1" style="max-width:500px;overflow-wrap:break-word">
                  <div style="min-height:24px"> <span style="font-weight:700">HTTP Bearer</span>  </div>
               </td>
               <td
                  colspan="3">
                  Send <code>Authorization</code> in <code>header</code> containing the word <code>Bearer</code> followed by a space
                  and then the JWT. 
                  <form style="display:flex"> 
                    <input id="username" type="text" spellcheck="false" class="api-key-input fs-exclude" data-hj-suppress data-sl="mask" value placeholder="username">
                    <input id="password" type="password" spellcheck="false" class="api-key-input fs-exclude" data-hj-suppress data-sl="mask" value placeholder="password">
                    <button type="submit" class="m-btn thin-border" style="margin-left:5px" part="btn btn-outline">SET</button> 
                  </form>
               </td>
            </tr>
         </table>
      </div>
   </slot>
</section>
```

Then you can hook up the button click to the javascript to grab the properties from your input form and pass those to the function: (or something like this

```js
const username = document.getByElementId('username').value;
const password = document.getByElementId('password').value;
document.getElementsByTagName("openapi-explorer")[0].setAuthenticationConfiguration('basicAuth', { token: `${username}:${password}' });
```

**Comment by splitbrain** [comment]
Ah I see. TBH that seems even worse, as I am duplicating something you already did. And I won't be benefiting from any improvement you might make to those forms in the future.
